pkg/connector/client/user.go

```diff
@@ -73,6 +74,39 @@ func (c *SalesforceClient) UserExist(
 	return user != nil, nil
 }
 
+func (c *SalesforceClient) GetUserByEmailWithRetry(
+	ctx context.Context,
+	email string,
+) (
+	*SalesforceUser,
+	error,
+) {
+	maxRetries := 3
+	baseDelay := time.Second
+	var err error
+
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		// Clear cache or we won't get new responses
+		err = uhttp.ClearCaches(ctx)
+		if err != nil {
+			return nil, err
+		}
+		user, err := c.GetUserByEmail(ctx, email)
+		if err == nil {
+			return user, nil
+		}
+
+		if status.Code(err) != codes.NotFound {
+			return nil, err
+		}
+		// Wait before retrying with exponential backoff
+		delay := time.Duration(attempt+1) * baseDelay
+		time.Sleep(delay)
```

Comment:
This feels like it should be handled by ticket processing with a well known return type and not here. This isnt durable imo.